### ISSUE TITLE
Metric without range doesn't work

### ISSUE THREAD

**Issue by maartendra** [laravel/nova-issues#301]
When using the default `ranges` for a Metric everything works well. But, when you just want to display the total count, without date range, it doesn't. The following code displays `0 No data`, even though there are a few rows.

```php
<?php

namespace App\Nova\Metrics;

use App\Customer;
use Illuminate\Http\Request;
use Laravel\Nova\Metrics\Value;

class TotalCustomers extends Value
{
    /**
     * Calculate the value of the metric.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return mixed
     */
    public function calculate(Request $request)
    {
        return $this->count($request, Customer::class);
    }

    /**
     * Get the URI key for the metric.
     *
     * @return string
     */
    public function uriKey()
    {
        return 'new-customers';
    }
}
```

It seems to go wrong within this function, but not sure how to fix:

```php
    /**
     * Prepare the metric for JSON serialization.
     *
     * @return array
     */
    public function jsonSerialize()
    {
        return array_merge(parent::jsonSerialize(), [
            'ranges' => collect($this->ranges() ?? [])->map(function ($range, $key) {
                return ['label' => $range, 'value' => $key];
            })->values()->all(),
        ]);
    }
```

**Comment by damian-pf** [comment]
You can pass your own values directly to the `result` function

```
/**
     * Calculate the value of the metric.
     *
     * @return mixed
     */
    public function calculate()
    {
        return $this->result(Customer::count());
    }
```

[This is the brief section covering it in the docs](https://nova.laravel.com/docs/1.0/metrics/defining-metrics.html#value-metrics)

**Comment by bkilshaw** [comment]
Can we get this re-opened? I get that there's a solution for it, but it doesn't work as you would expect.

The docs (https://nova.laravel.com/docs/3.0/metrics/defining-metrics.html#value-ranges)  say:

> Of course, you are not required to define any ranges at all:

While true, you can return an empty array or remove the function completely, it doesn't work as you would expect. My expectation was that if you removed all the ranges, it would just return the total count, but it doesn't, it still filters the results.

The docs need to be updated to explicitly state that not returning a range results in it using the default range (whatever that is).

Alternatively, what I would prefer to see is that not returning any ranges should result in counting the total records without filtering the date at all, but that would be a breaking change.

Aside, if the solution above is the recommended method to return a count of all records, could we at least get that added to the docs?

**Comment by roadsend** [comment]
You can overwrite the currentRange function to set a default.
I'm setting a 'range' with a month filter - which conflicted -- setting the range to return an empty array and using this function to override the calculated version worked -  for the next 10 years or so.

```
    protected function currentRange($range, $timezone)
    {
        return [
            now()->subDays(3600, ),
            now(),
        ];
    }
```

**Comment by tricic** [comment]
Any updates on this? It's still the same, when I remove all ranges I expected result not be date filtered, but it is. The simplest solution that I could think of is adding default range PHP_INT_MAX => 'All Time' which obviously is not ideal but works.
I simply want metrics range not to be filtered by any date range by default, because I'm filtering it with nemrutco/nova-global-filter package and I'm getting conflicts.

**Comment by davidhemphill** [comment]
👋 Hi there! In order to keep this repository focused on bug reports, we auto-close feature requests and requests for help. Feel free to post your feature requests so others can discuss and add reactions. We'll keep an eye on them for later planning.

**Comment by crynobone** [comment]
This has been implemented on v3.20.0

**Comment by muinmundzir** [comment]
> 
> 
> This has been implemented on v3.20.0

Correct me but I read releases for v3.20.0, they just added 'ALL' helper for value metrics, not for trend metrics.  I'm  still strugling to set default trend metrics to all time

> Added new ALL helper option for filtering Value metrics.
